File: DevExpress VCL 23.2.6 Full Source/VCL/ExpressGantt Control/Demos/CBuilder/Common/DemoBasicMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "DemoBasicMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxContainer"
#pragma link "cxControls"
#pragma link "cxDropDownEdit"
#pragma link "cxEdit"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma link "cxMaskEdit"
#pragma link "cxSpinEdit"
#pragma link "cxTextEdit"
#pragma link "dxCore"
#pragma link "dxGanttControl"
#pragma link "dxGanttControlAssignments"
#pragma link "dxGanttControlCustomClasses"
#pragma link "dxGanttControlCustomSheet"
#pragma link "dxGanttControlResources"
#pragma link "dxGanttControlTasks"
#pragma link "dxGanttControlViewChart"
#pragma link "dxGanttControlViewResourceSheet"
#pragma link "dxGanttControlViewTimeline"
#pragma link "dxLayoutContainer"
#pragma link "dxLayoutControl"
#pragma link "dxLayoutcxEditAdapters"
#pragma link "dxLayoutLookAndFeels"
#pragma resource "*.dfm"
#include "DemoUtils.h"
TDemoBasicMainForm *DemoBasicMainForm;
//---------------------------------------------------------------------------
__fastcall TDemoBasicMainForm::TDemoBasicMainForm(TComponent* Owner)
	: TdxForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TDemoBasicMainForm::FileExitExecute(TObject *Sender)
{
  Close();
}
//---------------------------------------------------------------------------
void __fastcall TDemoBasicMainForm::New1Click(TObject *Sender)
{
  if (EnableDocumentOpen())
  {
	GanttControl->DataModel->Reset();
	FModified = true;
  }
}
//---------------------------------------------------------------------------
void __fastcall TDemoBasicMainForm::Export1Click(TObject *Sender)
{
  if (SaveDialog->Execute())
  {
	GanttControl->SaveToFile(SaveDialog->FileName);
	FModified = false;
  }
}
//---------------------------------------------------------------------------
void __fastcall TDemoBasicMainForm::Import1Click(TObject *Sender)
{
  if (EnableDocumentOpen() && OpenDialog->Execute())
	GanttControl->LoadFromFile(OpenDialog->FileName);
}
//---------------------------------------------------------------------------
void __fastcall TDemoBasicMainForm::miAboutClick(TObject *Sender)
{
  ShowAboutDemoForm();
}
//---------------------------------------------------------------------------
void __fastcall TDemoBasicMainForm::FormCreate(TObject *Sender)
{
  SetDefaultLookAndFeel();
  AddLookAndFeelMenu();
  WindowState = wsMaximized;
}
//---------------------------------------------------------------------------
void TDemoBasicMainForm::AddLookAndFeelMenu()
{
  mmMain->Items->Insert(mmMain->Items->IndexOf(miAbout),
    CreateLookAndFeelMenuItems(mmMain->Items, lfController));
}
//---------------------------------------------------------------------------
TcxLookAndFeelKind TDemoBasicMainForm::GetDefaultLookAndFeelKind()
{
  return lfUltraFlat;
}
//---------------------------------------------------------------------------
bool TDemoBasicMainForm::EnableDocumentOpen()
{
  if (!FModified)
    return true;

  int mr = MessageDlg("Do you want to save your changes?", mtConfirmation, mbYesNoCancel, 0);
  if (mr == mrCancel)
    return false;
  bool Result = (mr == mrNo);
  if (!Result) 
  {
    Result = SaveDialog->Execute();
    if (Result)
      GanttControl->DataModel->SaveToFile(SaveDialog->FileName);
  }
  return Result;
}
//---------------------------------------------------------------------------
bool TDemoBasicMainForm::IsNativeDefaultStyle()
{
  return false;
}
//---------------------------------------------------------------------------
void TDemoBasicMainForm::SetDefaultLookAndFeel()
{
//  lfController->NativeStyle = IsNativeDefaultStyle;
//  lfController->Kind = GetDefaultLookAndFeelKind;
}
//---------------------------------------------------------------------------
void __fastcall TDemoBasicMainForm::lchbDirectXClick(TObject *Sender)
{
  if (lchbDirectX->Checked)
	lfController->RenderMode = rmDirectX;
  else
	lfController->RenderMode = rmGDI;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::cbSrcrollBarsPropertiesEditValueChanged(TObject *Sender)
{
  GanttControl->LookAndFeel->ScrollbarMode = TdxScrollbarMode(cbSrcrollBars->ItemIndex);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::lgActiveViewTabChanged(TObject *Sender)
{
  if (lgActiveView->ItemIndex == 0)
	GanttControl->ViewChart->Active = true;
  else if (lgActiveView->ItemIndex == 1)
	GanttControl->ViewResourceSheet->Active = true;
  else
    GanttControl->ViewTimeline->Active = true;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::lchbChartCellAutoHeightClick(TObject *Sender)
{
  GanttControl->ViewChart->OptionsSheet->CellAutoHeight = lchbChartCellAutoHeight->Checked;
  GanttControl->ViewChart->OptionsSheet->AllowColumnHide = lchbChartColumnHide->Checked;
  GanttControl->ViewChart->OptionsSheet->AllowColumnMove = lchbChartColumnMove->Checked;
  GanttControl->ViewChart->OptionsSheet->AllowColumnSize = lchbChartColumnSize->Checked;
  GanttControl->ViewChart->OptionsSheet->ColumnQuickCustomization = lchbChartColumnQuickCustomization->Checked;
  GanttControl->ViewChart->OptionsSheet->AllowColumnInsert = lcbChartColumnInsert->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::lchbChartVisibleClick(TObject *Sender)
{
  GanttControl->ViewChart->OptionsSheet->Visible = lchbChartVisible->Checked;
  lchbChartCellAutoHeight->Enabled = lchbChartVisible->Checked;
  lchbChartColumnHide->Enabled = lchbChartVisible->Checked;
  lchbChartColumnMove->Enabled = lchbChartVisible->Checked;
  lchbChartColumnSize->Enabled = lchbChartVisible->Checked;
  lchbChartColumnQuickCustomization->Enabled = lchbChartVisible->Checked;
  lcbChartColumnInsert->Enabled = lchbChartVisible->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::cmbChartTimescalePropertiesChange(TObject *Sender)
{
  GanttControl->ViewChart->TimescaleUnit = TdxGanttControlChartViewTimescaleUnit(cmbChartTimescale->ItemIndex);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::lchbResourceSheetCellAutoHeightClick(TObject *Sender)
{
  GanttControl->ViewResourceSheet->OptionsSheet->CellAutoHeight = lchbResourceSheetCellAutoHeight->Checked;
  GanttControl->ViewResourceSheet->OptionsSheet->AllowColumnHide = lchbResourceSheetColumnHide->Checked;
  GanttControl->ViewResourceSheet->OptionsSheet->AllowColumnMove = lchbResourceSheetColumnMove->Checked;
  GanttControl->ViewResourceSheet->OptionsSheet->AllowColumnSize = lchbResourceSheetColumnSize->Checked;
  GanttControl->ViewResourceSheet->OptionsSheet->ColumnQuickCustomization = lchbResourceSheetColumnQuickCustomization->Checked;
  GanttControl->ViewResourceSheet->OptionsSheet->AllowColumnInsert = lcbResourceSheetColumnInsert->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::lchbShowOnlyExplicitlyAddedTasksClick(TObject *Sender)
{
  GanttControl->ViewTimeline->ShowOnlyExplicitlyAddedTasks = lchbShowOnlyExplicitlyAddedTasks->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::cmbTimelineScalePropertiesChange(TObject *Sender)
{
  GanttControl->ViewTimeline->TimescaleUnit = TdxGanttControlTimelineViewTimescaleUnit(cmbTimelineScale->ItemIndex);
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::seTimelineUnitMinWidthPropertiesChange(TObject *Sender)
{
  GanttControl->ViewTimeline->TimescaleUnitMinWidth = seTimelineUnitMinWidth->Value;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::lchbDeleteConfirmClick(TObject *Sender)
{
  GanttControl->OptionsBehavior->ConfirmDelete = lchbDeleteConfirm->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::GanttControlActiveViewChanged(TObject *Sender)
{
  if (GanttControl->ViewChart->Active)
	lgActiveView->ItemIndex = 0;
  else if (GanttControl->ViewResourceSheet->Active)
	lgActiveView->ItemIndex = 1;
  else
	lgActiveView->ItemIndex = 2;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::GanttControlDataModelLoaded(TObject *Sender)
{
  FModified = false;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::GanttControlAssignmentChanged(TObject *Sender, TdxGanttControlAssignment *AAssignment)
{
  FModified = true;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::GanttControlResourceChanged(TObject *Sender, TdxGanttControlResource *AResource)
{
  FModified = true;
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::GanttControlTaskChanged(TObject *Sender, TdxGanttControlTask *ATask)
{
  FModified = true;
}
//---------------------------------------------------------------------------
void __fastcall TDemoBasicMainForm::aRedoExecute(TObject *Sender)
{
  GanttControl->History->Redo();
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::aRedoUpdate(TObject *Sender)
{
  aRedo->Enabled = GanttControl->History->CanRedo();
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::aUndoExecute(TObject *Sender)
{
  GanttControl->History->Undo();
}
//---------------------------------------------------------------------------

void __fastcall TDemoBasicMainForm::aUndoUpdate(TObject *Sender)
{
  aUndo->Enabled = GanttControl->History->CanUndo();
}
//---------------------------------------------------------------------------

File: DevExpress VCL 23.2.6 Full Source/VCL/ExpressGantt Control/Demos/Delphi/Common/DemoBasicMain.pas
unit DemoBasicMain;

interface

{$I cxVer.inc}

uses
{$IFDEF DELPHI16}
  System.UITypes,
{$ENDIF}
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms,
  Dialogs, Menus, ComCtrls, StdCtrls,
  dxCore, cxContainer, cxEdit, cxTextEdit, cxDropDownEdit, cxMaskEdit, cxSpinEdit,
  cxGraphics, cxClasses, cxControls, dxForms,
{$IFDEF EXPRESSSKINS}
  dxSkinsdxRibbonPainter,
  dxSkinsdxBarPainter,
{$ENDIF}
{$IFDEF EXPRESSBARS}
  dxBar, dxStatusBar,
{$ENDIF}
  cxLookAndFeels, cxLookAndFeelPainters, dxLayoutControl,
  dxLayoutContainer, dxLayoutLookAndFeels, dxLayoutcxEditAdapters, dxBarBuiltInMenu,
  dxGanttControlCustomClasses, dxGanttControl, dxGanttControlCustomSheet, dxGanttControlViewChart,
  dxGanttControlViewResourceSheet, dxGanttControlViewTimeline, dxGanttControlTasks, dxGanttControlAssignments,
  dxGanttControlResources, cxGroupBox, dxLayoutControlAdapters, cxButtons,
  ActnList, ImgList, cxImageList, dxShellDialogs;

type
  TDemoBasicMainForm = class(TdxForm)
    mmMain: TMainMenu;
    miAbout: TMenuItem;
    miFile: TMenuItem;
    miExit: TMenuItem;
    Separator1: TMenuItem;
    lfController: TcxLookAndFeelController;
    cxGroupBox1: TcxGroupBox;
    lcMainGroup_Root: TdxLayoutGroup;
    lcMain: TdxLayoutControl;
    GanttControl: TdxGanttControl;
    dxLayoutItem1: TdxLayoutItem;
    Export1: TMenuItem;
    Import1: TMenuItem;
    SaveDialog: TdxSaveFileDialog;
    OpenDialog: TdxOpenFileDialog;
    dxLayoutLookAndFeelList1: TdxLayoutLookAndFeelList;
    dxLayoutGroup1: TdxLayoutGroup;
    lgOptions: TdxLayoutGroup;
    lgActiveView: TdxLayoutGroup;
    lchbDeleteConfirm: TdxLayoutCheckBoxItem;
    lchbDirectX: TdxLayoutCheckBoxItem;
    lchbChartCellAutoHeight: TdxLayoutCheckBoxItem;
    lchbChartColumnHide: TdxLayoutCheckBoxItem;
    lchbChartColumnMove: TdxLayoutCheckBoxItem;
    lchbChartColumnSize: TdxLayoutCheckBoxItem;
    lchbChartColumnQuickCustomization: TdxLayoutCheckBoxItem;
    lchbChartVisible: TdxLayoutCheckBoxItem;
    dxLayoutGroup2: TdxLayoutGroup;
    dxLayoutGroup3: TdxLayoutGroup;
    dxLayoutGroup4: TdxLayoutGroup;
    dxLayoutGroup5: TdxLayoutGroup;
    lchbResourceSheetCellAutoHeight: TdxLayoutCheckBoxItem;
    lchbResourceSheetColumnHide: TdxLayoutCheckBoxItem;
    lchbResourceSheetColumnMove: TdxLayoutCheckBoxItem;
    lchbResourceSheetColumnSize: TdxLayoutCheckBoxItem;
    lchbResourceSheetColumnQuickCustomization: TdxLayoutCheckBoxItem;
    lchbShowOnlyExplicitlyAddedTasks: TdxLayoutCheckBoxItem;
    dxLayoutItem4: TdxLayoutItem;
    seTimelineUnitMinWidth: TcxSpinEdit;
    dxLayoutItem3: TdxLayoutItem;
    cmbTimelineScale: TcxComboBox;
    dxLayoutSeparatorItem1: TdxLayoutSeparatorItem;
    dxLayoutEmptySpaceItem1: TdxLayoutEmptySpaceItem;
    dxLayoutItem5: TdxLayoutItem;
    cbSrcrollBars: TcxComboBox;
    dxLayoutGroup6: TdxLayoutGroup;
    dxLayoutGroup7: TdxLayoutGroup;
    dxLayoutItem2: TdxLayoutItem;
    cmbChartTimescale: TcxComboBox;
    dxLayoutCxLookAndFeel1: TdxLayoutCxLookAndFeel;
    New1: TMenuItem;
    Separator2: TMenuItem;
    lbDescription: TLabel;
    lcbChartColumnInsert: TdxLayoutCheckBoxItem;
    lcbResourceSheetColumnInsert: TdxLayoutCheckBoxItem;
    dxLayoutGroup8: TdxLayoutGroup;
    cxButton1: TcxButton;
    dxLayoutItem6: TdxLayoutItem;
    cxButton2: TcxButton;
    dxLayoutItem7: TdxLayoutItem;
    ilActions: TcxImageList;
    alToolBar: TActionList;
    aUndo: TAction;
    aRedo: TAction;
    procedure FileExitExecute(Sender: TObject);
    procedure miAboutClick(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Export1Click(Sender: TObject);
    procedure Import1Click(Sender: TObject);
    procedure lchbDeleteConfirmClick(Sender: TObject);
    procedure lchbDirectXClick(Sender: TObject);
    procedure cbSrcrollBarsPropertiesEditValueChanged(Sender: TObject);
    procedure lgActiveViewTabChanged(Sender: TObject);
    procedure lchbChartCellAutoHeightClick(Sender: TObject);
    procedure lchbChartVisibleClick(Sender: TObject);
    procedure cmbChartTimescalePropertiesChange(Sender: TObject);
    procedure lchbResourceSheetCellAutoHeightClick(Sender: TObject);
    procedure lchbShowOnlyExplicitlyAddedTasksClick(Sender: TObject);
    procedure cmbTimelineScalePropertiesChange(Sender: TObject);
    procedure seTimelineUnitMinWidthPropertiesChange(Sender: TObject);
    procedure GanttControlActiveViewChanged(Sender: TObject);
    procedure New1Click(Sender: TObject);
    procedure GanttControlDataModelLoaded(Sender: TObject);
    procedure GanttControlAssignmentChanged(Sender: TObject; AAssignment: TdxGanttControlAssignment);
    procedure GanttControlResourceChanged(Sender: TObject; AResource: TdxGanttControlResource);
    procedure GanttControlTaskChanged(Sender: TObject; ATask: TdxGanttControlTask);
    procedure aUndoUpdate(Sender: TObject);
    procedure aRedoUpdate(Sender: TObject);
    procedure aUndoExecute(Sender: TObject);
    procedure aRedoExecute(Sender: TObject);
  protected
    FModified: Boolean;
    procedure AddLookAndFeelMenu; virtual;
    function EnableDocumentOpen: Boolean; virtual;
    function GetDefaultLookAndFeelKind: TcxLookAndFeelKind; virtual;
    function IsNativeDefaultStyle: Boolean; virtual;
    procedure SetDefaultLookAndFeel; virtual;
  public
  {$IFDEF EXPRESSBARS}
    BarManager: TdxBarManager;
  {$ENDIF}
  end;

var
  DemoBasicMainForm: TDemoBasicMainForm;

implementation

uses
  Variants, DateUtils, AboutDemoForm, SkinDemoUtils;

{$R *.dfm}

procedure TDemoBasicMainForm.FormCreate(Sender: TObject);
begin
  SetDefaultLookAndFeel;
  AddLookAndFeelMenu;
end;

procedure TDemoBasicMainForm.AddLookAndFeelMenu;
begin
{$IFDEF EXPRESSBARS}
  BarManager := TdxBarManager.Create(Self);
  dxBarConvertMainMenu(mmMain, BarManager);
  BarManager.Style := bmsUseLookAndFeel;
{$ENDIF}
{$IFDEF EXPRESSBARS}
  CreateSkinsMenuItem(BarManager);
{$ELSE}
  CreateSkinsMenuItem(mmMain);
{$ENDIF}
end;

function TDemoBasicMainForm.GetDefaultLookAndFeelKind: TcxLookAndFeelKind;
begin
  Result := lfUltraFlat;
end;

function TDemoBasicMainForm.IsNativeDefaultStyle: Boolean;
begin
  Result := False;
end;

procedure TDemoBasicMainForm.SetDefaultLookAndFeel;
begin
  lfController.NativeStyle := IsNativeDefaultStyle;
  lfController.Kind := GetDefaultLookAndFeelKind;
end;

procedure TDemoBasicMainForm.miAboutClick(Sender: TObject);
begin
  ShowAboutDemoForm;
end;

function TDemoBasicMainForm.EnableDocumentOpen: Boolean;
var
   mr: Integer;
begin
  Result := not FModified;
  if not Result then
  begin
    mr := MessageDlg('Do you want to save your changes?', mtConfirmation, mbYesNoCancel, 0);
    if mr = mrCancel then
      Exit;
    Result := mr = mrNo;
    if not Result then
    begin
      Result := SaveDialog.Execute;
      if Result then
        GanttControl.DataModel.SaveToFile(SaveDialog.FileName);
    end;
  end;
end;

procedure TDemoBasicMainForm.New1Click(Sender: TObject);
begin
  if EnableDocumentOpen then
  begin
    GanttControl.DataModel.Reset;
    FModified := False;
  end
end;

procedure TDemoBasicMainForm.Import1Click(Sender: TObject);
begin
  if EnableDocumentOpen and OpenDialog.Execute then
    GanttControl.LoadFromFile(OpenDialog.FileName);
end;

procedure TDemoBasicMainForm.Export1Click(Sender: TObject);
begin
  if SaveDialog.Execute then
  begin
    GanttControl.SaveToFile(SaveDialog.FileName);
    FModified := False;
  end;
end;

procedure TDemoBasicMainForm.FileExitExecute(Sender: TObject);
begin
  Close;
end;

procedure TDemoBasicMainForm.GanttControlActiveViewChanged(Sender: TObject);
begin
  if GanttControl.ViewChart.Active then
    lgActiveView.ItemIndex := 0
  else
  if GanttControl.ViewResourceSheet.Active then
    lgActiveView.ItemIndex := 1
  else
    lgActiveView.ItemIndex := 2;
end;

procedure TDemoBasicMainForm.GanttControlDataModelLoaded(Sender: TObject);
begin
  FModified := False;
end;

procedure TDemoBasicMainForm.GanttControlAssignmentChanged(Sender: TObject; AAssignment: TdxGanttControlAssignment);
begin
  FModified := True;
end;

procedure TDemoBasicMainForm.GanttControlResourceChanged(Sender: TObject; AResource: TdxGanttControlResource);
begin
  FModified := True;
end;

procedure TDemoBasicMainForm.GanttControlTaskChanged(Sender: TObject; ATask: TdxGanttControlTask);
begin
  FModified := True;
end;

procedure TDemoBasicMainForm.lgActiveViewTabChanged(Sender: TObject);
begin
  case lgActiveView.ItemIndex of
    0:
      GanttControl.ViewChart.Active := True;
    1:
      GanttControl.ViewResourceSheet.Active := True;
    2:
      GanttControl.ViewTimeline.Active := True;
  end;
end;

procedure TDemoBasicMainForm.cmbChartTimescalePropertiesChange(Sender: TObject);
begin
  GanttControl.ViewChart.TimescaleUnit := TdxGanttControlChartViewTimescaleUnit(cmbChartTimescale.ItemIndex);
end;

procedure TDemoBasicMainForm.lchbChartCellAutoHeightClick(Sender: TObject);
begin
  GanttControl.ViewChart.OptionsSheet.CellAutoHeight := lchbChartCellAutoHeight.Checked;
  GanttControl.ViewChart.OptionsSheet.AllowColumnHide := lchbChartColumnHide.Checked;
  GanttControl.ViewChart.OptionsSheet.AllowColumnInsert := lcbChartColumnInsert.Checked;
  GanttControl.ViewChart.OptionsSheet.AllowColumnMove := lchbChartColumnMove.Checked;
  GanttControl.ViewChart.OptionsSheet.AllowColumnSize := lchbChartColumnSize.Checked;
  GanttControl.ViewChart.OptionsSheet.ColumnQuickCustomization := lchbChartColumnQuickCustomization.Checked;
end;

procedure TDemoBasicMainForm.lchbChartVisibleClick(Sender: TObject);
begin
  GanttControl.ViewChart.OptionsSheet.Visible := lchbChartVisible.Checked;
  lchbChartCellAutoHeight.Enabled := lchbChartVisible.Checked;
  lchbChartColumnHide.Enabled := lchbChartVisible.Checked;
  lcbChartColumnInsert.Enabled := lchbChartVisible.Checked;
  lchbChartColumnMove.Enabled := lchbChartVisible.Checked;
  lchbChartColumnSize.Enabled := lchbChartVisible.Checked;
  lchbChartColumnQuickCustomization.Enabled := lchbChartVisible.Checked;
end;

procedure TDemoBasicMainForm.lchbResourceSheetCellAutoHeightClick(Sender: TObject);
begin
  GanttControl.ViewResourceSheet.OptionsSheet.CellAutoHeight := lchbResourceSheetCellAutoHeight.Checked;
  GanttControl.ViewResourceSheet.OptionsSheet.AllowColumnHide := lchbResourceSheetColumnHide.Checked;
  GanttControl.ViewResourceSheet.OptionsSheet.AllowColumnInsert := lcbResourceSheetColumnInsert.Checked;
  GanttControl.ViewResourceSheet.OptionsSheet.AllowColumnMove := lchbResourceSheetColumnMove.Checked;
  GanttControl.ViewResourceSheet.OptionsSheet.AllowColumnSize := lchbResourceSheetColumnSize.Checked;
  GanttControl.ViewResourceSheet.OptionsSheet.ColumnQuickCustomization := lchbResourceSheetColumnQuickCustomization.Checked;
end;

procedure TDemoBasicMainForm.lchbShowOnlyExplicitlyAddedTasksClick(Sender: TObject);
begin
  GanttControl.ViewTimeLine.ShowOnlyExplicitlyAddedTasks := lchbShowOnlyExplicitlyAddedTasks.Checked;
end;

procedure TDemoBasicMainForm.cmbTimelineScalePropertiesChange(Sender: TObject);
begin
  GanttControl.ViewTimeLine.TimescaleUnit := TdxGanttControlTimeLineViewTimescaleUnit(cmbTimelineScale.ItemIndex);
end;

procedure TDemoBasicMainForm.aRedoExecute(Sender: TObject);
begin
  GanttControl.History.Redo;
end;

procedure TDemoBasicMainForm.aRedoUpdate(Sender: TObject);
begin
  aRedo.Enabled := GanttControl.History.CanRedo;
end;

procedure TDemoBasicMainForm.aUndoExecute(Sender: TObject);
begin
  GanttControl.History.Undo;
end;

procedure TDemoBasicMainForm.aUndoUpdate(Sender: TObject);
begin
  aUndo.Enabled := GanttControl.History.CanUndo;
end;

procedure TDemoBasicMainForm.cbSrcrollBarsPropertiesEditValueChanged(Sender: TObject);
begin
  GanttControl.LookAndFeel.ScrollbarMode := TdxScrollbarMode(cbSrcrollBars.ItemIndex);
end;

procedure TDemoBasicMainForm.seTimelineUnitMinWidthPropertiesChange(Sender: TObject);
begin
  GanttControl.ViewTimeLine.TimescaleUnitMinWidth := seTimelineUnitMinWidth.Value;
end;

procedure TDemoBasicMainForm.lchbDeleteConfirmClick(Sender: TObject);
begin
  GanttControl.OptionsBehavior.ConfirmDelete := lchbDeleteConfirm.Checked;
end;

procedure TDemoBasicMainForm.lchbDirectXClick(Sender: TObject);
const
  AMode: array[Boolean] of TdxRenderMode = (rmGDI, rmDirectX);
begin
  lfController.RenderMode := AMode[lchbDirectX.Checked];
end;

initialization
  UseLatestCommonDialogs := False;

end.

